plugins/inventory/proxmox.py

```diff
@@ -506,7 +508,11 @@ def _handle_item(self, node, ittype, item):
         if item['status'] == 'stopped':
             self.inventory.add_child(self._group('all_stopped'), name)
         elif item['status'] == 'running':
-            self.inventory.add_child(self._group('all_running'), name)
+            item_status = 'running'
+            # get more details about the status of the qemu VM if want_facts == True
+            if want_facts:
+                item_status = properties.get(self._fact('qmpstatus'), item_status)
+            self.inventory.add_child(self._group('all_%s' % (item_status)), name)
```

Comment:
Hmm, isn't this a breaking change, since some hosts are now in different groups than before?

Reply:
Now they would be correct. Any VM that has a Paused state cannot be used and cannot be connected to. The reason for this was that Proxmox says that the VM is running, but in actuality is not. Without this, you have a wrong status for your machine. 

The only other thing I can think of is introduce a new group prefixed with qemu like `proxmox_qemu_paused` and leave it in running. That way it will always be added to running even though it's in a paused/preflight state

Reply:
We are gonna end up with the following groups. And we can do intersection between proxmox_all_qemu and proxmox_all_qemu_running to figure out which ones are currently actually running.
```
  |--@proxmox_all_lxc:
  |--@proxmox_all_qemu:
  |--@proxmox_all_qemu_paused:
  |--@proxmox_all_qemu_prelaunch:
  |--@proxmox_all_qemu_running:
  |--@proxmox_all_qemu_stopped:
  |--@proxmox_all_running:
  |--@proxmox_all_stopped:
```

Reply:
Adding the new groups (and not modifying the old ones) definitely works; the chance that it breaks an existing setup should be negligible.

What do the other proxmox maintainers think? Do you prefer the old solution, or tend more to this one (which is definitely backwards compatible)?

Reply:
I've changed it, so we still have the old behavior, and we just add them to the new group.

Reply:
I agree with @felixfontein that's is breaking, but in this case I'd say it's breaking for the better, as the current situation allows for errors when trying to run Ansible against these machines as they are unreachable. On the other hand, with those extra groups it's easy to take that in account, but I'm a bit worried it might also make the general use of it more complex (as running doesn't _always_ mean running)

@ilijamt Thanks for your work btw :)

Reply:
I'm fine with both, though I prefer the one where we just remove it from running and add it to the paused/prelaunched group, without the addition of the extra groups. 

The reason for this was because as you said, @Thulium-Drake the machines got paused during live migration and ansible tried to run and failed on the paused machines, which led me to restarting the process afterwards. As you said, if it's in _running_ group, doesn't mean it's actually running.

Let me know what you guys think, about which way to do this. I'm fine with both.

Reply:
If this would have been merged for 5.0.0 I wouldn't mind that much if it's a breaking change, but now 5.0.0 has passed and this is going to end up in 5.x.0. Breaking changes have to wait for 6.0.0 (i.e. in ~half a year).

How about having a new option which allows to decide in which groups not-really-running hosts end up, with default set to the current default? Then we can deprecate the default after some time, and eventually switch it to a new one somewhen in the future. That will prevent a sudden breaking change for folks depending on the current behavior, and everyone who wants to have the new behavior can have that already right now (by setting the option to the value that will become the default).

(The main disadvantage is that this would mean yet another option that eventually isn't needed anymore...)

Reply:
I'm inclined to still move ahead with it. @felixfontein absolutely right it's a breaking change on how it behaves, _however_ the current behavior should more be considered a bug then that this is a new feature. As @ilijamt pointed out, in his current setup, which might impact other users as well, the module breaks play executions when a KVM is currently moving somewhere else, because it's reporting as running, while it technically not is.

Reply:
It's not the only case, you can also manually suspend a VM, and it will report as running when it's suspended.

Reply:
So what is the consensus? cc @Thulium-Drake @felixfontein 

- Add a new flag to allow for the new statuses, and deprecate it on the new release. 
- Fix the issue now, and the statuses will be extended with the additional statuses `prelaunch` and `paused`

Reply:
I'm good with either, but I think option A is easier for @felixfontein to release right now :)

Reply:
If the next release would be a major release I would be OK with option B, but since we just had one it's probably better to go with option A. This feels too close to a breaking change to me, even though I can understand very well why the current behavior sucks :)